Bottom: c3e979f1b0895390878bca4ee8c8b6603795709e
Top:    ac59b2681fe3d5fb9826ecb148257410aed12cc0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-03 23:35:47 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index a122028..d89a968 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -230,6 +230,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
